lib/connection_pool/timed_stack.rb

```diff
@@ -169,9 +170,9 @@ def fetch_connection(options = nil)
   def shutdown_connections(options = nil)
     while connection_stored?(options)
       conn = fetch_connection(options)
+      @created -= 1 unless @created == 0
       @shutdown_block.call(conn)
```

Comment:
Should we rescue this block call? If the user code raises, will the pool orphan connections?

Reply:
> If the user code raises, will the pool orphan connections?

In both `#shutdown` and `#reap`, we will potentially orphan the connection that we run the `block.call` on.

However, I think its best that we consider the connection not owned by the ConnectionPool once we pass it to the `shutdown_block` or block passed to `reap`. It is up to the user at the point to properly handle closing the connection.

The other connections that are still in the pool can be closed by calling `shutdown` again if the user would like to ensure the remaining connections are closed.

>Should we rescue this block call?

As stated above, `shutdown` and `reap` can always be called again by the user if they would like to handle the case where the `block` raises an error.

I can write a test for this case for the above, proving we can call `#shutdown` and `#reap` again, if the first time raised an error.

Reply:
https://github.com/mperham/connection_pool/pull/192/commits/8eac79fde4875a5cef1a9d11e45e765afe552f1b Test Cases showing we can call #shutdown and #reap again, after an error!